src/com/google/enterprise/adaptor/database/TupleReader.java

```diff
@@ -205,45 +289,53 @@ protected void emitRow(ResultSet resultSet, ContentHandler handler)
         case Types.CHAR:
         case Types.VARCHAR:
           String string = resultSet.getString(col);
-          handler.startElement("", columnName, columnName, atts);
-          outWriter.write(string);
-          break;
-        case Types.CLOB:
-        case Types.LONGVARCHAR:
-          Reader reader = resultSet.getCharacterStream(col);
-          handler.startElement("", columnName, columnName, atts);
-          copyValidXMLCharacters(reader, outWriter);
-          reader.close();
+          if (string != null) {
+            try (Writer writer =
+                new BufferedWriter(new ContentHandlerWriter(handler))) {
+              handler.startElement("", columnName, columnName, atts);
+              writer.write(string);
+            }
+          }
           break;
         case Types.NCHAR:
-        case Types.NCLOB:
         case Types.NVARCHAR:
         case Types.LONGNVARCHAR:
-          Reader nReader = resultSet.getNCharacterStream(col);
-          handler.startElement("", columnName, columnName, atts);
-          copyValidXMLCharacters(nReader, outWriter);
-          nReader.close();
+          try (Reader reader = resultSet.getNCharacterStream(col);
+              Writer writer =
+                  new BufferedWriter(new ContentHandlerWriter(handler))) {
+            if (reader != null) {
+              handler.startElement("", columnName, columnName, atts);
+              copyValidXMLCharacters(reader, writer);
+            }
+          }
           break;
         case Types.BIT:
         case Types.BOOLEAN:
           Boolean value = resultSet.getBoolean(col);
-          handler.startElement("", columnName, columnName, atts);
-          outWriter.write(String.valueOf(value));
+          if (value != null) {
+            try (Writer writer =
+                new BufferedWriter(new ContentHandlerWriter(handler))) {
+              handler.startElement("", columnName, columnName, atts);
+              writer.write(String.valueOf(value));
+            }
+          }
           break;
         case Types.OTHER:
         default:
           string = resultSet.getString(col);
           if (null == string) {
             string = "" + resultSet.getObject(col);
+          } else {
```

Comment:
What? I think calling getObject is silly here, but you're changing the behavior of null here, too, but with strange code. Let's figure out the null handling, and then fix this.